docs/authentication-contribute.md

```diff
@@ -0,0 +1,8 @@
+# Authentication
+
+## Adding a new authentication
+
+To add a new authentication method:
+* Fork the repo and create a new branch
+* Code the implementation, making sure to expose the same interface to current versions.
```

Comment:
* Create the implementation in a similar method to how `google_auth.py` functions